### ISSUE TITLE
Allow building vscode with Python 3 (instead of Python 2)

### ISSUE THREAD

**Issue by jakobjw** [microsoft/vscode#115760]
The ["How to Contribute" site](https://github.com/microsoft/vscode/wiki/How-to-Contribute) says that for building vscode, Python >= 2.7 is required and "version 3 is not supported".

On many systems used as a build environment, it get's increasingly difficult to accept Python 2 to be installed and/or used, due to its already elapsed [end of life](https://www.python.org/doc/sunset-python-2/), which implies considerable security concerns.

The request is to change the build tooling in a way to use Python 3 instead of Python 2.

(Sorry if this is already discussed somewhere else, I searched and wasn't able to find anything.)

**Comment by yume-chan** [comment]
I believe Python 2.7 was required only for node-gyp (to compile node native addons)

However node-gyp has added support for Python 3.x long time ago. I think I only have Python 3.x but I can still build Code from source.

So maybe only this documentation is outdated.

**Comment by joaomoreno** [comment]
Give 3 a try. If it works, fantastic. If it doesn't, let us know what the issue is exactly. Our build and dev machines are more than happy to keep running 2.

**Comment by jakobjw** [comment]
Thanks, I went ahead and was able to **successfully build** vscode with solely **Python 3.8** on my system (macOS 11).

Before that, I tried it with solely Python 3.9 and got an error during the "node-gyp rebuild" command (as indicated by @yume-chan, node-gyp seems to be the reason for the Python version restriction).

These experiences seem to correspond with the [stated node-gyp requirements](https://github.com/nodejs/node-gyp) ("Python v2.7, v3.5, v3.6, v3.7, or v3.8").

Important notice in order not to get "deluded": after the successful build with Python 3.8, a build with solely Python 3.9 was also successful. However, this was most probably only the case because node-gyp has already been built before. I tried it on a second machine and confirmed this.

I would suggest to change the ["How to Contribute" site](https://github.com/microsoft/vscode/wiki/How-to-Contribute) in a way that it does not state any Python versions; instead there should be a reference to the [node-gyp requirements](https://github.com/nodejs/node-gyp) for allowing the reader to determine what are the current supported Python versions.
